tests/test_client_functional.py

```diff
@@ -3041,3 +3041,126 @@ async def handler(request):
         assert resp.status == 200
         assert await resp.text() == "ok"
         assert resp.headers["Content-Type"] == "text/plain; charset=utf-8"
+
+
+async def test_max_field_size_session_default(aiohttp_client: Any) -> None:
+    async def handler(request):
+        return web.Response(headers={"Custom": "x" * 8190})
+
+    app = web.Application()
+    app.add_routes([web.get("/", handler)])
+
+    client = await aiohttp_client(app)
+
+    async with await client.get("/") as resp:
+        assert resp.headers["Custom"] == "x" * 8190
+
+
+@pytest.mark.xfail
```

Comment:
`xfail` generally means the test is not working yet and requires something to be fixed.

This should be rewritten to test the exact expected behaviour (e.g. using `with pytest.raises(...)`).